src/script/conversation/MessageRepository.ts

```diff
@@ -786,6 +786,12 @@ export class MessageRepository {
     }
   }
 
+  public onReactionClick(conversationEntity: Conversation, message_et: ContentMessage, reaction: string): void {
+    if (!conversationEntity.removed_from_conversation()) {
+      window.setTimeout(() => this.sendReaction(conversationEntity, message_et, reaction), 100);
+    }
+  }
+
```

Comment:
@atomrc I am not sure about that, its the same function that we used for toggleLike I used here. I had to change the reaction type to be string. since it was an existing working code so i didn't change it

Reply:
/**
   * Toggle like status of message.
   *
   * @param conversationEntity Conversation entity
   * @param message_et Message to react to
   */
  public toggleLike(conversationEntity: Conversation, message_et: ContentMessage): void {
    if (!conversationEntity.removed_from_conversation()) {
      const reaction = message_et.is_liked() ? ReactionType.NONE : ReactionType.LIKE;
      message_et.is_liked(!message_et.is_liked());

      window.setTimeout(() => this.sendReaction(conversationEntity, message_et, reaction), 100);
    }
  }

Reply:
I'd say the timeout def doesn't belong here. I'd like to know why we need it and if needed we need to document this. 
This is not explicit why it's needed 🤔